### ISSUE TITLE
Missing types for Html5QrcodeScanner

### ISSUE THREAD

**Issue by kawpii** [briosheje/react-html5-qrcode-reader#3]
Currently `Html5QrcodeScanner` have `any` type, anyway we can have Typescript type for that?

**Comment by briosheje** [comment]
As far as I know there is no typings for the original library: if you know there are some, please let me know so that I will update this library properly.

**Comment by kawpii** [comment]
> As far as I know there is no typings for the original library: if you know there are some, please let me know so that I will update this library properly.

You are correct, I think still you can define `.d.ts` in your lib is not it?

**Comment by briosheje** [comment]
> 

It would be unreliable because if the original library somehow changes, I would need to manually update the types.

I will check whether there is something that can be done to automatically generate types from a specific library, although that probably should be done by the original package first intead.

**Comment by briosheje** [comment]
Update: it looks like the original library is written in typescript: https://github.com/mebjas/html5-qrcode/blob/master/src/html5-qrcode.ts

I will investigate this, perhaps the library has been rewritten and hence can be directly wired in and provide the typings.

This topic will be updated while I investigate stuff.
